ls1intum/Artemis
# `Assessment`: Focus inline feedback editor when reviewing programming submissions

### Checklist
#### General



#### Client

### Motivation and Context


Resolves #6675 
Once a tutor inserts inline feedback inside programming exercise manual assessments, you want to start writing immediately. Therefore the textbox should be focused in edit mode, which was not the case.

### Description
There are two cases in which we want the textbox to be focused: 
- *initial open*: the angular components for each line are created at page load and get visible through an added class name (this is part of the ace-code-editor functionality). Therefore, we can not solve this inside the angular component but focus the textbox when the class name is added externally.
- *enter edit mode on existing feedback*: we can do this inside the angular component by reacting to the `editFeedback` "event"

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Manual assessment

1. Log in to Artemis
2. Navigate to Course Administration > "your-exercise" > Exercise Dashboard > Start Assessment
3. Add new inline feedback and verify that the cursor is already inside the feedback textbox
4. Save the inline feedback
5. Click the `Edit` button and verify that the cursor is again inside the feedback textbox

#### Exam Mode Testing




### Review Progress






#### Code Review
#### Manual Tests


### Test Coverage







### Screenshots



https://github.com/ls1intum/Artemis/assets/78978542/319079c6-b6ca-486c-8c93-c6d217a1a733
